### PR TITLE
removed extra EnableAnalyzers props

### PR DESCRIPTION
**Pull request by vlada-shubina**
fixes https://github.com/dotnet/sdk/issues/27735

### CODE REVIEW

**Review by MichaelSimons — APPROVED** [review]

File: src/Cli/Microsoft.TemplateEngine.Cli/Microsoft.TemplateEngine.Cli.csproj [hunk 1]
```diff
@@ -4,7 +4,6 @@
     <Description>Template creation for the dotnet CLI</Description>
     <IsPackable>true</IsPackable>
     <IsShippingPackage>false</IsShippingPackage>
-    <EnableAnalyzers>true</EnableAnalyzers>
```
Comment (GangWang01) [inline src/Cli/Microsoft.TemplateEngine.Cli/Microsoft.TemplateEngine.Cli.csproj:7]: It is not extra. It enables analyzers specified in `eng/Analyzers.props` for build in development. For sourcebuild, it needs to overwrite the value when `DotNetBuildFromSource` is true in template engine projects.
```
<EnableAnalyzers>true</EnableAnalyzers>
<EnableAnalyzers Condition="'$(DotNetBuildFromSource)' == 'true'">false</EnableAnalyzers>
```

**Comment by vlada-shubina** [comment]
okay, I fixed it and enabled `Microsoft.CodeAnalysis.CSharp.CodeStyle` solution wide.
There were only few violations and I fixed them.
codeowners, @dotnet/dotnet-cli  please shout if you are not okay with this change,

**Comment by vlada-shubina** [comment]
I decided to move to `<EnforceCodeStyleInBuild>` property instead adding the reference - it looks simpler.

**Comment by vlada-shubina** [comment]
`CodeStyle` analyzer started to have problems in VS. Might be related to https://github.com/dotnet/sdk/pull/27747.

@MichaelSimons please let me know if this is urgent. If so, I can do another PR fixing StyleCop only and removing `CodeStyle` until the issue is fixed.

**Comment by MichaelSimons** [comment]
> @MichaelSimons please let me know if this is urgent. If so, I can do another PR fixing StyleCop only and removing `CodeStyle` until the issue is fixed.

A fix is needed for source-build by RC2.  We can't ship RC2 with prebuilts.

**Comment by vlada-shubina** [comment]
https://github.com/dotnet/sdk/pull/27747 is merged, so merging this one too.

File: src/ApiCompat/Microsoft.DotNet.ApiCompatibility/IAssemblySymbolLoader.cs [hunk 2]
```diff
@@ -1,7 +1,9 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+#if NET472_OR_GREATER
```
Comment (ViktorHofer) [inline src/ApiCompat/Microsoft.DotNet.ApiCompatibility/IAssemblySymbolLoader.cs:4]: @vlada-shubina curious, why was this change needed?

Comment (vlada-shubina) [reply src/ApiCompat/Microsoft.DotNet.ApiCompatibility/IAssemblySymbolLoader.cs:4]: good catch. This directive can be removed. 
Strangely enough it is not shown as an error in full framework, only in .NET.

Comment (vlada-shubina) [reply src/ApiCompat/Microsoft.DotNet.ApiCompatibility/IAssemblySymbolLoader.cs:4]: I'll fix it in another PR shortly.

Comment (vlada-shubina) [reply src/ApiCompat/Microsoft.DotNet.ApiCompatibility/IAssemblySymbolLoader.cs:4]: Fixed in https://github.com/dotnet/sdk/pull/27795/commits/1b030c7591e7f866530fcc1f11cc8dce71d98387
